drivers/digital-io/max22196/max22196.h

```diff
@@ -0,0 +1,141 @@
+*******************************************************************************/
+#ifndef _MAX22196_H
+#define _MAX22196_H
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "no_os_spi.h"
+#include "no_os_gpio.h"
+#include "no_os_util.h"
+
+#define MAX22196_FRAME_SIZE		2
+
+#define MAX22196_CHANNELS		8
+
+#define MAX22196_DI_STATE_REG		0x00
+#define MAX22196_FAULT1_REG		0x01
+#define MAX22196_F1MASK_REG		0x02
+#define MAX22196_CFG_REG(x)		(0x03 + (x))
+#define MAX22196_GLOBALCFG_REG		0x0B
+#define MAX22196_LED_REG		0x0C
+#define MAX22196_FAULT2_REG		0x0D
+#define MAX22196_F2MASK_REG		0x0E
+#define MAX22196_START_STOP_REG		0x0F
+#define MAX22196_CNT_LSB_REG(x)		(0x10 + 2 * (x))
+#define MAX22196_CNT_MSB_REG(x)		(0x11 + 2 * (x))
+
+#define MAX22196_ADDR_MASK		NO_OS_GENMASK(7, 6)
+#define MAX22196_REG_ADDR_MASK		NO_OS_GENMASK(5, 1)
+#define MAX22196_RW_MASK		NO_OS_BIT(0)
+
+#define MAX22196_DI_STATE_MASK(x)	NO_OS_BIT(x)
+#define MAX22196_HITHR_MASK		NO_OS_BIT(7)
+#define MAX22196_SOURCE_MASK		NO_OS_BIT(6)
+#define MAX22196_CURR_MASK		NO_OS_GENMASK(5, 4)
+#define MAX22196_FLTEN_MASK		NO_OS_BIT(3)
+#define MAX22196_DELAY_MASK		NO_OS_GENMASK(2, 0)
+#define MAX22196_CNT_MASK(x)		NO_OS_BIT(x)
+#define MAX22196_CFG_MASK		(NO_OS_BIT(7) | NO_OS_GENMASK(5, 4))
+
+enum max22196_mode {
+	MAX22196_SINK_MODE,
+	MAX22196_SOURCE_MODE,
+};
+
+enum max22196_delay {
+	MAX22196_DELAY_50US,
+	MAX22196_DELAY_100US,
+	MAX22196_DELAY_400US,
+	MAX22196_DELAY_800US,
+	MAX22196_DELAY_1600US,
+	MAX22196_DELAY_3200US,
+	MAX22196_DELAY_128000US,
+	MAX22196_DELAY_20MS
+};
+
+enum max22196_curr {
+	MAX22196_HTL_MODE,
+	MAX22196_1X_CURRENT,
+	MAX22196_3X_CURRENT,
+	MAX22196_TTL_OP_OFF
+};
+
+struct max22196_init_param {
+	uint32_t chip_address;
+	struct no_os_spi_init_param *comm_param;
+	struct no_os_gpio_init_param *crc_param;
+	bool crc_en;
+};
+
+struct max22196_desc {
+	uint32_t chip_address;
+	struct no_os_spi_desc *comm_desc;
+	struct no_os_gpio_desc *crc_desc;
+	uint8_t buff[MAX22196_FRAME_SIZE + 1];
+	bool crc_en;
+};
+
+int max22196_reg_write(struct max22196_desc *, uint32_t, uint32_t);
+
+int max22196_reg_read(struct max22196_desc *, uint32_t, uint32_t *);
+
+int max22196_reg_update(struct max22196_desc *, uint32_t, uint32_t, uint32_t);
+
+int max22196_set_mode(struct max22196_desc *, uint32_t, enum max22196_mode);
+
+int max22196_chan_cfg(struct max22196_desc *, uint32_t, uint32_t,
+		      enum max22196_curr);
+
+int max22196_filter_set(struct max22196_desc *, uint32_t, uint32_t,
+			enum max22196_delay);
+
+int max22196_filter_get(struct max22196_desc *, uint32_t, uint32_t *,
+			enum max22196_delay *);
+
+int max22196_set_chan_cnt(struct max22196_desc *, uint32_t, uint8_t, uint8_t);
+
+int max22196_get_chan_cnt(struct max22196_desc *, uint32_t, uint8_t *,
+			  uint8_t *);
+
+int max22196_init(struct max22196_desc **, struct max22196_init_param *);
+
+int max22196_remove(struct max22196_desc *);
```

Comment:
missing APIs brief description, I see in max22190 there are present so would be nice to have them here as well